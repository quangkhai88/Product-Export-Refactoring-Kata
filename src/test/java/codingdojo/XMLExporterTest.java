package codingdojo;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.List;

import static codingdojo.SampleModelObjects.*;

class XMLExporterTest {

    @Test
    void testFullExport() {
        List<Order> list = Arrays.asList(RecentOrder, OldOrder);
        String xml = XMLExporter.exportFull(list);
        Approvals.verifyXml(xml);
    }

    @Test
    void testTaxExport() {
        List<Order> list = Arrays.asList(RecentOrder, OldOrder);
        String xml = XMLExporter.exportTaxDetails(list);
        Approvals.verifyXml(xml);
    }

    @Test
    void testFullExportWithEvent() {
        List<Order> list = Arrays.asList(RecentOrder, OldOrder);
        RecentOrder.addProducts(Masterclass);
        String xml = XMLExporter.exportFull(list);
        Approvals.verifyXml(xml);
    }

    @Test
    void testFullExportWithEventAndWeight() {
        List<Order> list = Arrays.asList(RecentOrder, OldOrder);
        RecentOrder.addProducts(Masterclass);
        String xml = XMLExporter.exportFull(list);
        Approvals.verifyXml(xml);
    }


    @Test
    void testExportStore() {
        List<Order> list = Arrays.asList(RecentOrder, OldOrder);
        String xml = XMLExporter.exportFull(list);
        Approvals.verifyXml(xml);
    }

    @Test
    void testExportHistory() {
        List<Order> list = Arrays.asList(RecentOrder, OldOrder);
        String xml = XMLExporter.exportFull(list);
        Approvals.verifyXml(xml);
    }
}
