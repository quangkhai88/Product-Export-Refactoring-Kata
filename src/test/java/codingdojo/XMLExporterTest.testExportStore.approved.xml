<?xml version="1.0" encoding="UTF-8"?><orders>
  <order id="100">
    <product id="EVENT02" stylist="Celeste Pulchritudo">
      <price currency="USD">149.99</price>
      Makeover
    </product>
    <product id="EVENT01" stylist="Celeste Pulchritudo">
      <price currency="USD">119.99</price>
      Eyeshadow Masterclass
    </product>
    <product id="EVENT01" stylist="Celeste Pulchritudo">
      <price currency="USD">119.99</price>
      Eyeshadow Masterclass
    </product>
  </order>
  <order id="200">
    <product id="LIPSTICK01" weight="30">
      <price currency="USD">14.99</price>
      Cherry Bloom
    </product>
  </order>
</orders>
